plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/Plugin.java

```diff
@@ -0,0 +1,77 @@
+package io.jenkins.tools.pluginmanager.impl;
+
+import java.util.jar.JarFile;
+
+public class Plugin {
+    private String name;
+    private String version;
+    private String url;
+    private String archiveFileName;
+    private JarFile jarFile;
+    private boolean isPluginOptional;
+
+
+    public Plugin(String name, String version, String url) {
+        this.name = name;
+        this.version = version;
+        this.url = url;
+        this.archiveFileName = new StringBuffer(name).append(".jpi").toString();
+
+    }
+
+    public Plugin(String name, String version, boolean isPluginOptional) {
+        this.name = name;
+        this.version = version;
+        this.isPluginOptional = isPluginOptional;
+        this.archiveFileName = new StringBuffer(name).append(".jpi").toString();
+    }
+
+    public void setName(String name) {
```

Comment:
is there a reason you have this in the constructor and a setter for it?
its also odd that you're setting archive file name at the same time, normally this sort of class is whats called a Pojo (plain old java object)
and calling a getter / setter will only do what you expect and nothing else

Reply:
So more or less I wanted the name of the plugin and its archive name to always be in sync. There was a note in the original bash script that " some plugin don't follow the rules about artifact ID (typically: docker-plugin)" and my understanding is that "-plugin" should be appended to the names of these plugins, which would update the plugin names and thus the archiveFileName should be updated at the same time.  I could call setArchiveFileName() in the PluginManager right after resetting the plugin name, but that seemed like it could lead to mistakes later down the line if someone forgets to do this. This is why I set it at the same time within the Plugin class.  But if you think this makes more sense, I'm happy to change it. Or maybe I should just remove archiveFileName as a field and add a method for getting it that returns the current pluginName + ".jpi"?

Reply:
removing the set archive file name method would make sense to me, good idea

Reply:
Resolved in 7ca3d5d987f6c60617f5f95994574849fb17f170.